Add capabilities to beats and agent comparison table

mostlyjason opened the pull request:
Add capabilities to beats and agent comparison table to highlight benefits of Elastic Agent and Fleet, and capabilities that only Beats supports

CC @nimarezainia

apmmachine commented:
A documentation preview will be available soon:</br><ul><li>📚 [HTML diff](https://observability-docs_943.docs-preview.app.elstc.co/diff)</li><li>📘 [Fleet User Guide](https://observability-docs_943.docs-preview.app.elstc.co/guide/en/fleet/master/index.html)</li><li>📙 [Observability Guide](https://observability-docs_943.docs-preview.app.elstc.co/guide/en/observability/master/index.html)</li></ui>
<!--COMMENT_GENERATED_WITH_ID_comment.id-->

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+
+|===
+
+Also, both Beats and {agent} standalone work well with version control, IaaC, and change management processes because the configuration is a simple text file. {fleet} only supports these processes through its REST API.
```
```suggestion
Also, both {beats} and {agent} standalone work well with version control, IaaC, and change management processes because the configuration is a simple text file. {fleet} only supports these processes through its REST API.
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
 |Supported
 |===
 
+[discrete]
```
I'm assuming this is just a move operation with no changes to the text?

mostlyjason replied:
@dedemorton I'm not sure what you mean? This is a whole new table. I just copied this line from the table above, but am I not supposed to?

dedemorton replied:
Ohhh. I need to slow down a little on reviews. I'll take a closer look.

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -217,7 +219,7 @@ configuration experience.
 
 |{filebeat-ref}/configuration-autodiscover.html[Autodiscover]
-|Autodiscover is facilitated through <<dynamic-input-configuration,dynamic inputs>>
+|Autodiscover is facilitated through <<dynamic-input-configuration,dynamic inputs>>. Elastic Agent does not support hints-based autodiscovery.
```
```suggestion
|Autodiscover is facilitated through <<dynamic-input-configuration,dynamic inputs>>. {agent} does not support hints-based autodiscovery.
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -133,12 +133,12 @@ NOTE: {endpoint-sec} has a different output matrix.
 |{es} Service
 |{y}
-|{y}
+|{y} only to the same cluster where Fleet runs
```
```suggestion
|{y} only to the same cluster where {fleet} runs
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -133,12 +133,12 @@ NOTE: {endpoint-sec} has a different output matrix.
 |{es}
 |{y}
-|{y}
+|{y} only to the same cluster where Fleet runs
```
```suggestion
|{y} only to the same cluster where {fleet} runs
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -167,6 +167,8 @@ NOTE: {endpoint-sec} has a different output matrix.
 |{n}
 |===
 
+Support for outputing to remote {es} clusters in {fleet}-managed agents is under consideration for a future release.
```
I had to read this a few times before it made sense. I think you're saying that Fleet-managed agents must send data to the same {es} cluster where {fleet} is running.

Maybe it would be good to state that up front and then say that support for sending to remote Elasticsearch clusters is under consideration for a future release. Or you can add your statement as a [footnote](https://docs.asciidoctor.org/asciidoc/latest/macros/footnote/) so that it's clear you are referring to the caveats listed in the table. Only problem with footnotes is that they end up on the very bottom of the page (rather than after the table).

mostlyjason replied:
@dedemorton I adjusted the wording here https://github.com/elastic/observability-docs/pull/943/commits/71333463664c275af3ac78de2583e6b9f1130a19 is that better?

dedemorton reviewed (commented):
mostly minor stuff.

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+[[additional-capabilties-beats-and-agent]]
+== Additional capabilities
+
+The following table shows the additional capabilties supported by Beats and the {agent} in {version}:
```
```suggestion
The following table shows the additional capabilities supported by {beats} and the {agent} in {version}:
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+
+[options,header]
+|===
+|Item |Beats |{fleet}-managed {agent} |Standalone {agent}
```
```suggestion
|Item |{beats} |{fleet}-managed {agent} |Standalone {agent}
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{y}
+|{y}
+
+|Configure from integrations UI
```
```suggestion
|Configure from Integrations UI
```

dedemorton replied:
It's correct for integrations to be lowercase except when you are describing the UI.

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{y}
+|{y}
+
+|Auto-generated ES API keys
```
```suggestion
|Auto-generated {es} API keys
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{y}
+|{n}
+
+|Auto-generate minimal ES privileges on the edge
```
```suggestion
|Auto-generate minimal {es} privileges on the edge
```

dedemorton commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|Requires superuser role
+|{n}
+|{y}
+|{n} (required to use integrations UI)
```
```suggestion
|{n} (required to use Integrations UI)
```

dedemorton reviewed (commented):
More comments. Sorry, I'm sure why my brain didn't realize these were additions. Mondays....

dedemorton reviewed (approved):
LGTM

dedemorton commented:
@mostlyjason Let me know if you want to backport this after you merge it.

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -167,6 +167,8 @@ NOTE: {endpoint-sec} has a different output matrix.
 |===
 
+Currently, {agent}s managed by Fleet can only output to the same {es} cluster where {fleet} is running. Support for outputting to remote {es} clusters is under consideration for a future release.
+
```
@mostlyjason if we don't have a public issue for it, then please create and link it here. It will be a great way to collect inputs from users on its priority

mostlyjason replied:
done

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+[[additional-capabilties-beats-and-agent]]
+== Additional capabilities
+
+The following table shows the additional capabilities supported by {beats} and the {agent} in {version}:
```
Please add a description column to describe each item. It helps users understand better what each item specifically does. For example - installing integration from UI -> This installs ES assets such as ES index templates, ingest pipelines, ILM policies and Kibana assets such as dashboards, saved views for discover etc.

mostlyjason replied:
done

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{y}
+|{y}
+
+|Auto-generated {es} API keys
```
Description should call out the benefits for generating API keys such as enforced restricted privileges

mostlyjason replied:
done

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{y}
+|{y}
+
+|Variables and input conditions
```
how does this work in fleet mode? Where do users specify these variables and input conditions and for what use cases?

mostlyjason replied:
done

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{y}
+|{y}
+
+|Requires superuser role to add integrations and agent polices
```
Requires superuser role to add integrations and agent polices from integration and fleet UIs respectively

mostlyjason replied:
done

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{y} 
+|{n} (it's optional to use the Integrations UI)
+
+|Air-gapped network support
```
Call out the specific components that need internet connection. For EPR, point to package registry repo on how to run offline EPR as an experimental feature and how to connect integrations UI to it.

mostlyjason replied:
done

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+|{n}
+|{y}
+
+|Mulitple outputs
```
do we have docs about beats and standalone supporting multiple outputs? I didn't find them. If we don't have docs, let's not call them supported. If we do, then link to the docs

mostlyjason replied:
Output docs are in progress https://github.com/elastic/beats/issues/23742. We can link them when they are available.

bmorelli25 replied:
Added a link.

mukeshelastic commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,106 @@ per output type, which enables loadbalancing.
+
+|===
+
+Also, both {beats} and {agent} standalone work well with version control, IaaC, and change management processes because the configuration is a simple text file. {fleet} only supports these processes through its REST API.
```
Can we link to fleet API docs?

mostlyjason replied:
We have an open issue to publish API docs here https://github.com/elastic/obs-dc-team/issues/308

mostlyjason replied:
I suggest adding a link after we publish the docs

mukeshelastic reviewed (dismissed):
Thank you so much @mostlyjason for writing these up. I was initially hesitant to publish these gaps because some of them are future feature functionalities. However, this doc goes in our spirit of openness and allows us to collect feedback from our users and community in a self service way. Appreciate the effort you put in writing these out.

I left few comments to make sure users understand what each item means and can provide feedback for missing gaps or future functionality. 

@elastic/integrations @elastic/agent @elastic/fleet fyi

mostlyjason commented:
@mukeshelastic I added another commit to incorporate your feedback as well as input from discussions with others. I think I addressed all your comments but please let me know if I'm missing anything.

mostlyjason commented:
@mukeshelastic @nimarezainia @dedemorton I'm going to be on PTO next week so I'll hand it off to you to review and make adjustments in the meantime. Mukesh suggested reviewing the descriptions to make sure they differentiate clearly between beats, fleet and standalone agents. I'd also appreciate input from Dede on copy and that I'm using the right asciidoc syntax.

dedemorton commented:
I'm on PTO next week, too. @EamonnTP or @bmorelli25 might be able to help while I'm away.

mukeshelastic commented:
@bmorelli25 I am blocking 30 mins to review this in person with you on Monday the 16th Aug.

mostlyjason commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,134 @@ per output type, which enables loadbalancing.
+|{n}
+|{y}
+|{y}
+|{agent}s use {ref}/data-streams.html[data streams] with easier index life cycle management and the https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme[data stream naming scheme]. {beats} uses a single index with potentially thousands of fields.
```
Looks like we have one here https://www.elastic.co/guide/en/fleet/7.15/data-streams.html

bmorelli25 replied:
Fixed

bmorelli25 commented on docs/en/ingest-management/beats-agent-comparison.asciidoc:
```diff
@@ -238,6 +240,134 @@ per output type, which enables loadbalancing.
+|{n}
+|{y}
+|{y}
+|{agent} integrations are installed in a convenient web UI or API, but {beats} modules are installed in a CLI. This installs {es} assets such as index templates and ingest pipelines, and {kib} assets such as dashboards.
```
Fixed, thanks
